test/integration/targets/hcloud_volume/tasks/main.yml

```diff
@@ -0,0 +1,165 @@
+    state: started
+    location: "fsn1"
+  register: main_server
+- name: verify setup server
+  assert:
+    that:
+      - main_server is changed
+
+- name: test missing size parameter on create Volume
+  hcloud_volume:
+    name: "{{hcloud_volume_name}}"
+    server: "{{hcloud_server_name}}"
+  register: result
+  ignore_errors: yes
+- name: verify fail test missing size parameter on create Volume
+  assert:
+    that:
+      - result is failed
+      - 'result.msg == "missing required arguments: size"'
+
+- name: test create volume with check mode
+  hcloud_volume:
+    name: "{{hcloud_volume_name}}"
+    size: 10
+    location: "fsn1"
+  register: result
+  check_mode: yes
+- name: verify create volume with check mode result
+  assert:
+    that:
+      - result is changed
+
+- name: test create volume
+  hcloud_volume:
+    name: "{{hcloud_volume_name}}"
+    size: 10
+    location: "fsn1"
+  register: volume
+- name: verify test create volume
+  assert:
+    that:
+      - volume is changed
+      - volume.hcloud_volume.name == "{{hcloud_volume_name}}"
+      - volume.hcloud_volume.location == "fsn1"
+      - volume.hcloud_volume.size == 10
+      - volume.hcloud_volume.server != "{{hcloud_server_name}}"
+
+- name: test create volume idempotence
+  hcloud_volume:
+    name: "{{hcloud_volume_name}}"
+    size: 10
+    location: "fsn1"
+  register: volume
+- name: verify test create volume
+  assert:
+    that:
+      - result is not changed
+
+- name: test attach volume with checkmode
+  hcloud_volume:
+    name: "{{hcloud_volume_name}}"
+    server: "{{hcloud_server_name}}"
+  check_mode: yes
+  register: volume
+- name: verify test attach volume with checkmode
+  assert:
+    that:
+      - volume is changed
+      - volume.hcloud_volume.server != "{{hcloud_server_name}}"
+
+- name: test attach volume
+  hcloud_volume:
+    name: "{{hcloud_volume_name}}"
+    server: "{{hcloud_server_name}}"
+  register: volume
+- name: verify attach volume
+  assert:
+    that:
+      - volume is changed
+      - volume.hcloud_volume.server == "{{hcloud_server_name}}"
+      - main_server is changed
+
+- name: test detach volume with checkmode
+  hcloud_volume:
+    name: "{{hcloud_volume_name}}"
+  check_mode: yes
+  register: volume
+- name: verify detach volume with checkmode
+  assert:
+    that:
+      - volume is changed
+      - volume.hcloud_volume.server == "{{hcloud_server_name}}"
+
+- name: test detach volume
+  hcloud_volume:
+    name: "{{hcloud_volume_name}}"
+  register: volume
+- name: verify detach volume
+  assert:
+    that:
+      - volume is changed
+      - volume.hcloud_volume.location == "fsn1"
+      - volume.hcloud_volume.server != "{{hcloud_server_name}}"
+      - main_server is changed
+
+- name: test update volume label
+  hcloud_volume:
+    name: "{{hcloud_volume_name}}"
+    labels:
+      key: value
+  register: volume
+- name: verify test update volume lable
+  assert:
+    that:
+      - volume is success
```

Comment:
this should be "is changed" right? I would also suggest to add tests to actually prove the label has the new value.